src/iris/api.py

```diff
@@ -3999,6 +4001,44 @@ def on_get(self, req, resp, app_name):
         resp.body = ujson.dumps(stats, sort_keys=True)
 
 
+def restrict_apps(req, resp, resource, params):
+    if req.context.get('app', {}).get('name') not in resource.allowed_apps:
+        raise HTTPForbidden('App not allowed to register devices')
+
+
+class Devices(object):
+    allow_read_no_auth = False
+
+    def __init__(self, config):
+        self.allowed_apps = config.get('devices_allowed_apps', [])
+
+    @before(restrict_apps)
+    def on_post(self, req, resp):
+        data = ujson.loads(req.context['body'])
+        user = data.get('username')
+        registration_id = data.get('registration_id')
+        platform = data.get('platform')
+
+        if user is None or registration_id is None or os is None:
+            raise HTTPBadRequest('Missing parameters for adding device')
+
+        # Open database connection
+        conn = db.engine.raw_connection()
+        cursor = conn.cursor()
+
+        cursor.execute('''INSERT IGNORE INTO device (registration_id, user_id, platform)
```

Comment:
Should do error handling here to clean up the DB connection and log exception during failures